Collections/collections_home_task.py

```diff
@@ -0,0 +1,458 @@
+"""
+Home task for Module 2 - Collections
+
+Write a code, which will:
+
+1. create a list of random number of dicts (from 2 to 10)
+
+dict's random numbers of keys should be letter,
+dict's values should be a number (0-100),
+example: [{'a': 5, 'b': 7, 'g': 11}, {'a': 3, 'c': 35, 'g': 42}]
+2. get previously generated list of dicts and create one common dict:
+
+if dicts have same key, we will take max value, and rename key with dict number with max value
+if key is only in one dict - take it as is,
+example: {'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+Each line of code should be commented with description.
+
+Commit script to git repository and provide link as home task result.
+"""
+import random  # Import of random for generation of random numbers and letters in dictionaries
+import string  # Import of string for operations with keys of dictionaries
+
+# Create a list of random number of dicts (from 2 to 10)
+list_of_random_dictionaries = []  # Create empty list
+for i in range(0, random.randint(2, 10)):  # FOR statement with randon number of iterations between 2 and 10
+    sample_dictionary = {random.choice(string.ascii_lowercase): random.randint(0, 100)
+                         for y in range(0, random.randint(0, 100))}  # Generation of random dictionary
```

Comment:
```suggestion
                         for y in range(random.randint(2, 3))}  # Generation of random dictionary
```
You can't use randint from 0 here, you will get empty dict in this case. I suggest to start with =randint(2, 3) and try to merge these dicts. When it works you will just extend this value to 100 and that's it. It's just easier for debugging to use small structures.

Reply:
You can also set up some bounded set of values, for instance ('a', 'b', 'c', 'd') and make choice as one of them instead of all ascii letters to make it easier for debugging. Logic will be the same.

Reply:
Fixed.